### DISCUSSION TITLE
컴포넌트 파일 구조를 어떻게 하는게 좋을까요??

### DISCUSSION THREAD

**Discussion by Jeong-Taeho** [prgrms-web-devcourse/Team-6Heroes-OneDayHero-FE#118] [Q&A]
컴포넌트 구조를 mission이 있다면 생성, 상세페이지 등등 페이지 별로 사용되는 컴포넌트가 다를 수 있으므로 route 구조에 따라 나눠버리는게 나을까요 아니면 misson이면 미션 관련 페이지에서 사용되는 컴포넌트들을 misson 폴더 아래 전부 사용하는 방식으로 하는게 좋을까요..? 나눴을 때는 깔끔하다는 장점이 있지만 비슷한 맥락의 페이지 안에서는 이리저리 같이 쓰이는 것도 많아서 고민되네요..

**Comment by howons** [comment]
개인적으론 처음엔 route 구조로 쓰다가 재사용이 필요할 때 차근차근 올리는 게 편할 것 같아요. 페이지 하나 구현할 때 컴포넌트의 재사용 가능성을 처음부터 따져보는 것 보다 실제로 재사용이 필요할 때 그동안 구현된 컴포넌트들 목록에서 찾아보는 게 쉽지 않을까 싶습니다!
